src/facade/dragonfly_connection.cc

```diff
@@ -745,33 +749,65 @@ void Connection::DispatchFiber(util::FiberSocketBase* peer) {
   DispatchOperations dispatch_op{builder, this};
   uint64_t request_cache_limit = absl::GetFlag(FLAGS_request_cache_limit);
 
+  size_t squashing_threshold = absl::GetFlag(FLAGS_pipeline_squash);
+
   while (!builder->GetError()) {
     evc_.await(
         [this] { return cc_->conn_closing || (!dispatch_q_.empty() && !cc_->sync_dispatch); });
     if (cc_->conn_closing)
       break;
 
-    MessageHandle msg = move(dispatch_q_.front());
-    dispatch_q_.pop_front();
+    auto recycle = [this, request_cache_limit](MessageHandle msg) {
+      dispatch_q_bytes_.fetch_sub(msg.UsedMemory(), memory_order_relaxed);
 
-    builder->SetBatchMode(dispatch_q_.size() > 0);
+      // Retain pipeline message in pool.
+      if (auto* pipe = get_if<PipelineMessagePtr>(&msg.handle); pipe) {
+        dispatch_q_cmds_count_--;
```

Comment:
Shouldn't this be atomic? Since `DispatchFiber` runs on a separate fiber?

Reply:
DispatchFiber always run on the same thread, all other variables are also not synchronized. Only the byte count is atomic, it's used to provide backpressure to other threads wanting to push